File: Branches/VBInicial/vpPriV100GrupoMundifios/FornecedoresCertificadosFilopa/Base/FichaCliente/BasIsFichaCliente.vb
Imports Primavera.Extensibility.Base.Editors
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports Primavera.Extensibility.Constants.ExtensibilityService
Imports StdBE100
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico

Namespace FornecedoresCertificadosFilopa

    Public Class BasIsFichaCliente
        Inherits FichaClientes

        'Public Overrides Sub TeclaPressionada(KeyCode As Integer, Shift As Integer, e As ExtensibilityEventArgs)
        '    MyBase.TeclaPressionada(KeyCode, Shift, e)

        '    If Module1.VerificaToken("FornecedoresCertificadosFilopa") = 1 Then

        '        If KeyCode = 81 And Me.Cliente.Inactivo = False Then
        '            Module1.certEntidade = Me.Cliente.Cliente


        '            Dim result As ExtensibilityResult = PriV100Api.BSO.Extensibility.CreateCustomFormInstance(GetType(FrmFornecedoresCertsFilopaView))

        '            If result.ResultCode = ExtensibilityResultCode.Ok Then

        '                Dim frm As FrmFornecedoresCertsFilopaView = result.Result
        '                frm.ShowDialog()

        '            End If
        '        End If

        '        If KeyCode = 81 And Me.Cliente.Inactivo = True Then
        '            MsgBox("Cliente Anulado! Não é possível abrir o formulário de certificados!", vbInformation)
        '        End If

        '    End If

        'End Sub

    End Class
End Namespace

File: Trunk/vpPriV100Filopa/FornecedoresCertificadosFilopa/Base/FichaCliente/BasIsFichaCliente.cs
using Generico;
using Primavera.Extensibility.Base.Editors;
using Primavera.Extensibility.BusinessEntities;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Constants.ExtensibilityService;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace FornecedoresCertificadosFilopa
{
    public class BasIsFichaCliente : FichaClientes
    {

        public override void TeclaPressionada(int KeyCode, int Shift, ExtensibilityEventArgs e)
        {
            base.TeclaPressionada(KeyCode, Shift, e);

            if (KeyCode == 81 & this.Cliente.Inactivo == false)
            {
                Module1.certEntidade = this.Cliente.Cliente;


                ExtensibilityResult result = BSO.Extensibility.CreateCustomFormInstance(typeof(FrmFornecedoresCertsFilopaView));

                if (result.ResultCode == ExtensibilityResultCode.Ok)
                {
                    FrmFornecedoresCertsFilopaView frm = result.Result;
                    frm.ShowDialog();
                }
            }

            if (KeyCode == 81 & this.Cliente.Inactivo == true)
                MessageBox.Show("Cliente Anulado! Não é possível abrir o formulário de certificados!", "", MessageBoxButtons.OK, MessageBoxIcon.Information);

        }


    }
}
